flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/aggfunctions/MaxAggFunctionWithRetract.scala

```diff
@@ -72,37 +72,34 @@ abstract class MaxWithRetractAggFunction[T](implicit ord: Ordering[T]) extends A
 
       a.f1 -= 1L
```

Comment:
I think we don't need the `f1: Long` field (total count). We can also check `f2.size() == 0`